chore: upgrade administrator to v6 api

jasont7 opened the pull request:
<!---
Describe your changes in a bulleted list.

Be sure to identify and justify breaking changes.
--->

## Pre-Review Checklist
* [x] I have considered backwards and forwards compatibility.
* [x] All changes are tested.
* [x] All touched code documentation is updated.
* [ ] All tests pass in my local environment
* [x] Deepsource issues have been reviewed and addressed.
* [x] Commented code and `console` usages have been removed.
 
## Screenshots
_Only required for visual changes_

jasont7 committed:
refactor: use v6 relative links, remove history & match prop

jasont7 committed:
refactor: replace switch with v6 routes

jasont7 committed:
fix: relative paths

jasont7 committed:
refactor: upgrade to react router v6

jasont7 committed:
chore: remove subtractions upgrade

jasont7 commented:
Tests currently failing:

`CreateUser`: seems to be a path resolution issue as the test html indicates a blank page. This is strange because a similar render in the `UserDetail` test does not give the same issue.

`UserDetail`: Only `UserGroups` section is failing - I suspect unrelated to path issues as before since other components of the `UserDetail` test are fine.

jasont7 committed:
fix: deepsource

ReeceHoffmann commented:
Just looking at the annotations GH is showing me, all of the errors look like they are getting stuck with a loader. Generally that is nock mocking problem. I would look for changes that could result in the mocked and used URLs having a mismatch?

ReeceHoffmann committed:
fix: ensure new routes work correctly

 - migrates createUser to V6 APIs
 - fixes tests
 - Correctly uses location from useLocation instead of window for setting search params

ReeceHoffmann committed:
Merge remote-tracking branch 'origin/main' into jason-thomo/ui-364-upgrade-administrator-to-v6-api
